models/issue_label.go

```diff
@@ -325,6 +330,129 @@ func GetLabelsByRepoID(repoID int64, sortType string, listOptions ListOptions) (
 	return getLabelsByRepoID(x, repoID, sortType, listOptions)
 }
 
+// ________
+// \_____  \_______  ____
+//  /   |   \_  __ \/ ___\
+// /    |    \  | \/ /_/  >
+// \_______  /__|  \___  /
+//         \/     /_____/
+
+// getLabelInOrgByName returns a label by Name in given organization
+// If pass orgID as 0, then ORM will ignore limitation of organization
+// and can return arbitrary label with any valid ID.
+func getLabelInOrgByName(e Engine, orgID int64, labelName string) (*Label, error) {
+	if len(labelName) == 0 {
+		return nil, ErrOrgLabelNotExist{0, orgID} // FIX ME!
+	}
+
+	l := &Label{
+		Name:  labelName,
+		OrgID: orgID,
+	}
+	has, err := e.Get(l)
```

Comment:
Many labels may match this pattern when orgID == 0. It will also bring labels not related to organizations. It's probably better to reject orgID == 0.

Reply:
Good call, have added checks for orgID >= 0 on all of these and matching unit tests too.